service_mesh/v2x/upgrading-ossm-to-2-1.adoc

```diff
@@ -0,0 +1,94 @@
+
+. Check your v2 `ServiceMeshControlPlane` resource configuration to make sure it is valid.
++
+.. Run the following command to view your `ServiceMeshControlPlane` resource as a v2 resource.
++
+[source,terminal]
+----
+$ oc get smcp -o yaml
+----
++
+.. Check the `spec.techPreview.errored.message` field in the output for information about any invalid fields.
++
+. Back up your control plane configuration. Switch to the project that contains your `ServiceMeshControlPlane` resource.  In this example, `istio-system` is the name of the control plane project.
++
+[source,terminal]
+----
+$ oc project istio-system
+----
++
+. Update the `.spec.version` field from v2.0 to v2.1.
++
+If you see the following message, then you need to change the existing "Mixer" type to “Istiod” type in the existing Control Plane spec before you update the  `.spec.version` field:
++
+An error occurred
+admission webhook `smcp.validation.maistra.io` denied the request: [support for `policy.type` "Mixer" and `policy.Mixer` options have been removed in v2.1, please use another alternative, support for `telemetry.type` "Mixer" and `telemetry.Mixer` options have been removed in v2.1, please use another alternative]”
++
+For example
++
+[source,terminal]
+----
+spec:
+  policy:
+    type: Istoid
+  telemetry:
+    type: Istiod
+  version: v2.1
+----
++
+Alternatively, you can use the console to create the control plane. In the {product-title} web console, click *Project*. Then, select the project name you just entered.
++
+.. Click *Operators* -> *Installed Operators*.
+.. Click *Create ServiceMeshControlPlane*.
+.. Select *YAML view* and update text of the YAML file, as shown in the previous example.
+.. Click *Save*.
++
+Wait until the reconciliation completes and check your `ServiceMeshMemberRoll`` Status Condition: Ready.
+
+[id="ossm-upgrading-differences-2-1_{context}"]
+== Configuring the 2.1 ServiceMeshControlPlane
+
+The `ServiceMeshControlPlane` resource has been changed for {ProductName} version 2.1. After you created a v2 version of the `ServiceMeshControlPlane` resource, modify it to take advantage of the new features and to fit your deployment. Consider the following changes to the specification and behavior of {ProductName} 2.1 as you're modifying your `ServiceMeshControlPlane` resource. You can also refer to the {ProductName} 2.1 product documentation for new information to features you use. The v2 resource must be used for {ProductName} 2.1 installations.
+
+[id="ossm-upgrading-differences-arch_{context}"]
+=== Architecture changes
+
+Although Mixer is no longer supported as a control plane component, Mixer policy and telemetry plugins are now supported through WASM extensions in Istiod.
+
+[id="ossm-upgrading-differences-behavior_{context}"]
+=== Behavioral changes
+
+* AuthorizationPolicy updates
+** If using PROXY protocol, updates are required if using ipBlocks/notIpBlocks, as the remote IPs are now specified using remoteIpBlocks/notRemoteIpBlocks
```

Comment:
```suggestion
** With the PROXY protocol, updates are required to use `ipBlocks` and `notIpBlocks`, because the remote IPs are now specified using `remoteIpBlocks` and `notRemoteIpBlocks`.
```

Reply:
Maybe
```suggestion
** With the PROXY protocol, if you're using `ipBlocks` and `notIpBlocks` to specify remote IP addresses, update the configuration to use `remoteIpBlocks` and `notRemoteIpBlocks` instead.
```

Reply:
Done